File: benchmarks/poly/fibonacci/fibonacci.py
import sys, time


def fib(n):
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)


def main():
    _t0 = time.perf_counter_ns()
    for i in range(10):
        print(fib(i))
    print(f"bench_ns={time.perf_counter_ns() - _t0}", file=sys.stderr)


if __name__ == "__main__":
    main()

File: benchmarks/poly/fibonacci/fibonacci.go
package main

import (
	"fmt"
	"os"
	"time"
)

func fib(n int) int {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	t0 := time.Now()
	for i := 0; i < 10; i++ {
		fmt.Println(fib(i))
	}
	fmt.Fprintf(os.Stderr, "bench_ns=%d\n", time.Since(t0).Nanoseconds())
}
